analysis_scripts/analysis_utils.R

```diff
@@ -213,3 +213,85 @@ compute_bootstrap_location <- function(full_data_df, year_col_id, article_col_id
     return(quantile_res)
 
 }
+
+
+#' Get the word frequencies from a JSON file containing
+#' file_ids and the body of an articles
+#'
+#' @param json_file This contains all the text from a specific year and news-type
+#' @param file_ids_pass optional parameter to only include specific file_ids
+#' @return word_freq a dataframe of word counts
+calc_word_freq <- function(json_file, file_ids_pass=NA){
+    
+        # read in the json
+        all_articles_df = read_json(json_file)
+        
+        # get the articles we are interested in
+        if(!is.na(file_ids_pass)){
+            all_articles_df = subset(all_articles_df, file_id %in% file_ids_pass)
+        }
+        all_articles_df = data.table(all_articles_df)
+        
+        # tokenize
+        all_articles_df <- all_articles_df %>%
+                unnest_tokens(word, body)
```

Comment:
WOW!! This is a really neat function. Python requires a bit more manual effort to get the same result.

Reply:
Yeah1 I followed this tutorial: https://www.tidytextmining.com/tidytext.html